build for win-arm64

trim21 opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

ring doesn't support us cl.exe to cross compile to arm64 yet

trim21 committed:
try

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

trim21 committed:
only keep win-arm64

trim21 committed:
try fix

trim21 committed:
I think it works

trim21 commented:
I think rust_win-arm64 fooget set CARGO_TARGET to aarch64-pc-windows-msvc for corss compile

pavelzw commented:
https://github.com/conda-forge/rust-activation-feedstock/blob/main/recipe/activate.bat Needs to be adjusted

trim21 committed:
Merge branch 'main' into win-arm64

trim21 commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15517877906](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15517877906). Examine the logs at this URL for more detail.</sub>

trim21 commented:
> i'm wondering why we don't need to adjust anything in https://github.com/conda-forge/rust-activation-feedstock/blob/main/recipe/activate.bat 🤔

I'm not sure what I should do here... should I move `set CARGO_TARGET=aarch64-pc-windows-msvc` part to `rust-activation`?

pavelzw commented:
ah i didn't see that you adjusted this.

yes, that would be nice. these environment variables should belong there and not in each feedstock.

see https://github.com/conda-forge/rust-activation-feedstock/blob/a2a3d9fe93ad4cd3d61cc148791e122f886f05d9/recipe/activate.sh#L9 on how it's done for linux. we need to do something similar for rust

trim21 commented:
OK so there are 2 separated issues here. the `CARGO_TARGET` is a rust-activation upstream issue, and the need of clang is caused by ring, which is a rust crate upstream issue and that belong to this repo

pavelzw commented:
@wolfv do you need clang upstream in rattler-build as well or are you able to use msvc there?

chawyehsu commented:
So what is preventing this from being merged? I see CI were all green?

trim21 commented:
OK I see someone has done it https://github.com/conda-forge/rust-activation-feedstock/pull/90

I'll remove the workaround and lets see if ci passed

trim21 committed:
Update recipe.yaml

trim21 committed:
Merge branch 'main' into win-arm64

trim21 committed:
Update recipe.yaml

trim21 commented:
@conda-forge-admin, please rerender

trim21 commented:
Now I have done all I can, re-request a review from @pavelzw and @wolfv and (@baszalmstra maybe)

pavelzw commented on recipe/recipe.yaml:
```diff
@@ -48,7 +55,7 @@ requirements:
   - script:
       - rattler-build --help
-    # check that `bin/generate-cli-docs` is not present
+      # check that `bin/generate-cli-docs` is not present
```
```suggestion
    # check that `bin/generate-cli-docs` is not present
```

otherwise there will be merge conflicts with #192

trim21 committed:
Update recipe/recipe.yaml

Co-authored-by: Pavel Zwerschke <pavelzw@gmail.com>

pavelzw commented:
> Not really necessary I think, Previously we do not have win-arm64 build so it will still upload a new win-arm64 package

~~it might still produce different build strings for the other platforms which would lead to two `*_0` builds per platform. not sure whether the conditional dependencies will be included in the build strings for the other platforms~~ 

just checked it doesn't but i guess it also doesn't hurt to bump the numnber

trim21 commented on recipe/recipe.yaml:
```diff
@@ -17,6 +17,9 @@ build:
         - unset CARGO_TARGET_X86_64_APPLE_DARWIN_LINKER
+    - if: target_platform == "win-arm64" 
+      then: 
+        - set CC=clang 
```
it's a cross compile and a crate need clang in this case. which we talked before https://github.com/conda-forge/rattler-build-feedstock/pull/158#issuecomment-2955136445

trim21 replied:
we already have it at build required sections `- clang # ring need it for cross-compiling to target aarch64-pc-windows-msvc`

pavelzw replied:
also, in general, i'm wondering whether we can get away without the need for ring. when not using rustls, ring might not be needed.  might be conflicting with the aws crates

trim21 committed:
bump build number

trim21 committed:
Update recipe.yaml

trim21 commented:
can we merge this now? Or we still need someone fixing the linux_ppc64le?

pavelzw commented:
fixing ppc64le is not necessary, see https://github.com/conda-forge/conda-forge.github.io/issues/2510#issuecomment-3719175532

But i would still like to look into https://github.com/conda-forge/rattler-build-feedstock/pull/158#discussion_r2675554207 before, i'll try to get that done today

pavelzw commented on recipe/recipe.yaml:
```diff
@@ -17,6 +17,9 @@ build:
         - unset CARGO_TARGET_X86_64_APPLE_DARWIN_LINKER
+    - if: target_platform == "win-arm64" 
+      then: 
+        - set CC=clang   # ring need it for cross-compiling to target aarch64-pc-windows-msvc
```
i think a better way of doing this would be to set `c_compiler: clang` in `recipe/conda_build_config.yaml`. for this, we would need to build `clang_win-arm64` which would ensure correct activation scripts. this would result in this env variable not being needed.

see https://conda-forge.org/news/2024/04/30/clang-everywhere

https://github.com/conda-forge/clang-win-activation-feedstock/issues/63

trim21 replied:
can we merge this and improve it later?

pavelzw committed:
use msvc

pavelzw committed:
MNT: Re-rendered with conda-smithy 3.54.2 and conda-forge-pinning 2026.02.01.23.20.06

Other tools:
- conda-build 26.1.0
- rattler-build 0.55.1
- rattler-build-conda-compat 1.4.10

pavelzw committed:
fix build

pavelzw committed:
Merge branch 'main' into win-arm64

pavelzw commented:
https://github.com/seanmonstar/reqwest/releases/tag/v0.13.0 reqwest doesn't use `ring` anymore but instead `aws-lc`. let's see whether this is okay being compiled with msvc

pavelzw committed:
Merge branch 'main' into win-arm64

pavelzw commented on recipe/recipe.yaml:
```diff
@@ -17,20 +17,25 @@ build:
         # use the default linker for macOS as we are hitting a bug with the conda-forge linker
         - unset CARGO_TARGET_X86_64_APPLE_DARWIN_LINKER
+    # TODO: remove after nativetls patch is gone
+    - cargo update
```
```suggestion
```

pavelzw committed:
no update

wolfv committed:
try aws-lc-sys no asm for win-arm64

pavelzw committed:
WIP

wolfv committed:
try 0.58.0
